src/transxchange/naptan.rs

```diff
@@ -267,4 +273,40 @@ mod tests {
             read_stop_areas(csv_content.as_bytes()).unwrap();
         }
     }
+
+    mod read_stop_in_area {
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        #[test]
+        fn parsing_works() {
+            let csv_content = r#""StopAreaCode","AtcoCode"
+"010G0005","01000053220"
+"910GBDMNSTR","0100BDMNSTR0""#;
+            let stops_in_area = read_stops_in_area(csv_content.as_bytes()).unwrap();
+            assert_eq!(stops_in_area.len(), 2);
+            let stop_area_code = stops_in_area.get("01000053220").unwrap();
+            assert_eq!(stop_area_code, "010G0005");
+            let stop_area_code = stops_in_area.get("0100BDMNSTR0").unwrap();
+            assert_eq!(stop_area_code, "910GBDMNSTR");
+        }
+
+        #[test]
+        #[should_panic]
+        fn no_atco_code() {
+            let csv_content = r#""StopAreaCode"
+"010G0005"
+"910GBDMNSTR""#;
+            read_stops_in_area(csv_content.as_bytes()).unwrap();
+        }
+
+        #[test]
+        #[should_panic]
+        fn no_stop_area_code() {
```

Comment:
Probably not in business use cases which means we probably we might need integration tests for it.

My philosophy about unit tests is to try to test every possible case that can happen in the code (because unit tests are cheap to execute). And add the valuable business test cases in integration tests.

But I don't have a very strong opinion about that so if you feel like I should remove those tests...

Reply:
for me you are testing the csv lib.
as you don't do anything with error returned by csv lib. This test is useless imho

Reply:
I guess that's fair. I need to test the [line 132](https://github.com/CanalTP/transit_model/pull/294/files#diff-9c970ed467ca42773ff2ca90a574e331R132) for failure but both [`no_atco_code`](https://github.com/CanalTP/transit_model/pull/294/files#diff-9c970ed467ca42773ff2ca90a574e331R296) and [`no_stop_area_code`](https://github.com/CanalTP/transit_model/pull/294/files#diff-9c970ed467ca42773ff2ca90a574e331R305) are doing that so I can probably keep only one of them. I'll remove `no_stop_area_code`.